### PR TITLE
Implement ConvertOptions and Value API

### PR DESCRIPTION
**Pull request by joshua-honig**
This fully implements the approved APIs for the `ConvertOptions`, `ConvertOptionsBuilder`, and associated classes:

 - [`ConvertOptions`](https://github.com/ockham-net/ockham.net.convert/blob/c63bf5150523b0163d8a03ef739718226d44f9c0/api/lib/ConvertOptions.cs)
 - [`BooleanConvertOptions`](https://github.com/ockham-net/ockham.net.convert/blob/c63bf5150523b0163d8a03ef739718226d44f9c0/api/lib/ConvertOptions.Booleans.cs)
 - [`EnumConvertOptions`](https://github.com/ockham-net/ockham.net.convert/blob/c63bf5150523b0163d8a03ef739718226d44f9c0/api/lib/ConvertOptions.Enums.cs)
 - [`NumberConvertOptions`](https://github.com/ockham-net/ockham.net.convert/blob/c63bf5150523b0163d8a03ef739718226d44f9c0/api/lib/ConvertOptions.Numbers.cs)
 - [`StringConvertOptions`](https://github.com/ockham-net/ockham.net.convert/blob/c63bf5150523b0163d8a03ef739718226d44f9c0/api/lib/ConvertOptions.Strings.cs)
 - [`ValueTypeConvertOptions`](https://github.com/ockham-net/ockham.net.convert/blob/c63bf5150523b0163d8a03ef739718226d44f9c0/api/lib/ConvertOptions.Values.cs)
 - [`ConvertOptionsBuilder`](https://github.com/ockham-net/ockham.net.convert/blob/c63bf5150523b0163d8a03ef739718226d44f9c0/api/lib/ConvertOptions.Builder.cs)

It also implements the API for the `Value` class:

 - [`Value`](https://github.com/ockham-net/ockham.net.convert/blob/c63bf5150523b0163d8a03ef739718226d44f9c0/api/lib/Value.cs)

All implemented APIs have 100% test coverage (per dotcover). I also set up the API comparison using GenAPI. [`_excludeMembers.txt`](https://github.com/ockham-net/ockham.net.convert/compare/feature/convert-options?expand=1#diff-1df24d95dd0f0ee8a60a8fc7b0cbee24) is a list of items in the full API to exclude from comparison, because they're not implemented yet.

### CODE REVIEW

File: src/lib/Internal/TypeInspection.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Reflection;
+
+namespace Ockham.Data
+{
+
+    /// <summary>
+    /// NOT part of the public API of this project. Type inspection per se is not the goal of the Ockham.NET.Convert module
+    /// </summary>
+    internal class TypeInspection
```
Comment (joshua-honig) [inline src/lib/Internal/TypeInspection.cs:10]: Note this class is internal and does not affect the public API as approved in #2, #3

File: src/lib/Ockham.Convert.csproj [hunk 2]
```diff
@@ -1,8 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net472;netcoreapp2.2</TargetFrameworks>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (joshua-honig) [inline src/lib/Ockham.Convert.csproj:4]: Just build for netstandard2.0, since it includes all necessary APIs

**Comment by joshua-honig** [comment]
Every publicly visible member now has a documentation comment, too.

**Comment by joshua-honig** [comment]
Unit test coverage:

image

File: src/lib/Internal/TypeInspection.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            }
+        }
+        */
+
+        /// <summary>
+        /// Determine if the specified type is a numeric (integer, float, or decimal) type. Returns true for enums.
+        /// </summary> 
+        public static bool IsNumberType(Type type)
+        {
+            return _IsNumberType(type, false);
```
Comment (kendelljoseph) [inline src/lib/Internal/TypeInspection.cs:50]: I'm guessing the `intergersOnly` flag is a way to prepare for future options there. 👍

File: src/lib/Options/ConvertOptions.Booleans.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Ockham.Data
+{
+    /// <summary>
+    /// Settings for conversion to boolean values. This class is immutable.
+    /// </summary>
+    public class BooleanConvertOptions : OptionSet
```
Comment (kendelljoseph) [inline src/lib/Options/ConvertOptions.Booleans.cs:10]: Love seeing `OptionSet` getting used here. 👍

**Review by kendelljoseph — APPROVED** [review]
Looks ready for production! 👍

**Pull request merged**